File: 0523. Continuous Subarray Sum.cpp
class Solution
{
public:
    bool checkSubarraySum(vector<int>& nums, int k)
    {
        map<int, int> m;
        m[0] = 0;
        int sum = 0;
        
        for (int i = 0; i < nums.size(); ++i)
        {
            sum += nums[i];
            
            if (m.find(sum % k) == m.end())
            {
                m[sum % k] = i + 1;
            }
            else if (m[sum % k] < i)
            {
                return true;
            }
        }
        
        return false;
    }
};

File: 0523. Continuous Subarray Sum.cs
public class Solution
{
    public bool CheckSubarraySum(int[] nums, int k)
    {
        var modIndexies = new Dictionary<int, int>
        {
            { 0, -1 }
        };
        var prefixMod = 0;

        for (int i = 0; i < nums.Length; ++i)
        {
            prefixMod = (prefixMod + nums[i]) % k;

            if (modIndexies.TryGetValue(prefixMod, out var modIndex))
            {
                if (i - modIndex > 1)
                {
                    return true;
                }
            }
            else
            {
                modIndexies[prefixMod] = i;
            }
        }

        return false;
    }
}